### PR TITLE
First version of the PWA client.

### PR DESCRIPTION
**Pull request by michelangelo**
In this early version we set up all the needed machinery to serve the
Progressive WebApp client. Most of the URLs had to be changed to be put
within the scope of the app.

When this change lands in production the clients [1] will be able to add
Emotly on the home screen; the app will be kinda usable also when the
user is offline or with bad data connection.

A lot of field testing will be necessary to define all the corner cases.

[1] ATM the only platform to support (almost) all the features we make
use of is Android/Chrome.

### CODE REVIEW

File: emotly/static/app/js/sw.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+self.addEventListener('activate', function(event) {
+    L('activate');
+
+    // Populate with eventual stuff to keep.
+    var cacheWhitelist = [EMOTLY_CACHE_NAME];
+
+    event.waitUntil(caches.keys().then(function(cacheNames) {
+        return Promise.all(cacheNames.map(function(cacheName) {
+            if (cacheWhitelist.indexOf(cacheName) === -1) {
+                console.log(cacheName + " is gonna be deleted");
```
Comment (tizianasellitto) [inline emotly/static/app/js/sw.js]: Why not use the L function?

File: requirements.txt [hunk 2]
```diff
@@ -1,6 +1,7 @@
 bcrypt==2.0.0
 cffi==1.5.2
 Flask==0.10.1
+Flask-JSON==0.3.0
```
Comment (tizianasellitto) [inline requirements.txt]: Remove if not used.

**Comment by tizianasellitto** [comment]
LGTM.

**Comment by michelangelo** [comment]
Thanks @tizianasellitto, I've addressed your comments and updated the branch.

**Comment by ggc87** [comment]
ok!

**Pull request merged**